### PR TITLE
Add wolfTPM wrapper support for NV

### PR DESCRIPTION
**Pull request by dgarske**
* Added new API's (`wolfTPM2_NVStoreKey` and `wolfTPM2_NVDeleteKey`) for evict control support for saving and deleting keys to NV.
* Added new API's (`wolfTPM2_NVCreate`, `wolfTPM2_NVWrite`, `wolfTPM2_NVRead` and `wolfTPM2_NVDelete`) for NV create, write, read and delete.
* Added new API `wolfTPM2_Clear` to clear (fully reset) TPM.
* Added new API wrappers (`wolfTPM2_LoadRsaPublicKey` and `wolfTPM2_LoadEccPublicKey`) for loading RSA and ECC public keys.
* Separated the demo code into native and wrapper test applications. `./examples/tpm/tpm2_demo` is now `./examples/native/native_test` and `./examples/wrap/wrap_test`.
* Moved the generic utility functions into native area to support `WOLFTPM2_NO_WRAPPER`.
* Added autoconf options for `--disable-wrapper` and `--disable-examples`.
* Fix for SPI device open on Raspbian Kernel 4.9.
* Updates to README.md.

### CODE REVIEW

File: examples/native/native_test.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* tpm2_demo.h
+/* native_test.h
```
Comment (JacobBarthelmeh) [inline examples/native/native_test.h]: file permisions changed from 100644 to 100755

File: examples/native/native_test.c [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* tpm2_demo.c
+/* native_test.c
```
Comment (JacobBarthelmeh) [inline examples/native/native_test.c]: file permisions changed

File: src/tpm2_wrap.c [hunk 3]
```diff
@@ -323,6 +409,94 @@ int wolfTPM2_ReadPublicKey(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* key,
...
+
+    /* Move key into NV to persist */
+    XMEMSET(&in, 0, sizeof(in));
+    in.auth = primaryHandle;
+    in.objectHandle = key->handle.hndl;
+    in.persistentHandle = persistentHandle;
+
+    rc = TPM2_EvictControl(&in);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("TPM2_EvictControl failed %d: %s\n", rc, wolfTPM2_GetRCString(rc));
```
Comment (JacobBarthelmeh) [inline src/tpm2_wrap.c]: It may be best with this and other printf calls to use something along the lines of XPRINTF instead?

Comment (dgarske) [reply src/tpm2_wrap.c]: Good suggestion, but I will defer to future PR. For now requiring stdlib printf is sufficient.

**Pull request merged**